### ISSUE TITLE
Implement some kind of file type / mime type on files

### ISSUE THREAD

**Issue by AstromechZA** [AstromechZA/bunkr#15]
At the moment there is no distinction between text files and binary files. Because binary files are REALLY BIG compared to text files, we should prevent users from opening them when they might cause issues.

We could use a simple file type "text" or "binary". This should be good enough until we can start supporting opening files in custom dialogs. Then we should be able to declare media types like "binary/pdf", "binary/jpg", "binary/png" or "text/code" which will declare specific ways of presenting the file.

When a file is imported, the mime type should be autodetected based on character set and length and maybe using the file extension as a hint. The type can then be overridden using a CLI call. In the GUI, once the type has been autodetected, it should prompt the user to select a specific type.

Then in the GUI, if the file is a binary file with no way of opening it, the user will be notified and the action cancelled.

**Comment by AstromechZA** [comment]
Mimetypes I'd like to be able to view in the GUI in the future:

- markdown (already implemented)
- html (should be the base class of the markdown viewer)
- pdf (depends on having a decent / secure PDF viewing component)
- jpg / png / common image formats
- source code (basic monospaced editor with line numbers)

Need to be careful about where to draw the line. Any other file types should be treated as application/octet-stream and should require Exporting them before they can be opened.

**Comment by AstromechZA** [comment]
I worked today on an alternate plain text editor in b11acbd..0353ea5 ended up reverting it in e6c0380. I realised that the first mime types I implemented (plaintext/markdown/code) were focused on **editing**, but in hindsight the focus of the mime types should be on **viewing** the data.

The only goal for **editing** should be markdown. It allows the user to write many different types of documents, including short sections of code if they want to using the ``` ```syntax``` ```.
